app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt

```diff
@@ -437,6 +438,26 @@ open class TripsViewModel(
         })
   }
 
+  /**
+   * Copies the currently selected trip and assigns the current user as its sole participant.
+   *
+   * This function retrieves the current user from the `UserViewModel`, extracts the user's ID, and
+   * assigns it to the list of participants of the selected trip. The updated trip object is then
+   * passed to the `createTrip` function to save the copied trip.
+   *
+   * @param userViewModel The [UserViewModel] that provides the current user data.
+   * @param onSuccess A callback invoked upon the successful creation of the copied trip.
+   */
+  fun copyTrip(userViewModel: UserViewModel, onSuccess: () -> Unit) {
+    userViewModel.user
+    val trip =
+        userViewModel.user.value
```

Comment:
Here use the id from `Firebase.auth` as all the user data might have not loaded yet as it's an async operation and the `Firebase.auth.uid` is sure to loaded / cached